packages/neptune-extension/src/nbextension/Toolbar.js

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+
+import {
+  injectNotebookId,
+  getNotebookId,
+} from 'platform/meta';
+
+const NEPTUNE_LOGO_URL = 'url(\'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAA4AAAANCAYAAACZ3F9/AAAABHNCSVQICAgIfAhkiAAAAN5JREFUKJHF0K9Lw0EcxvGXc07ZEJTJBIOrBothJsHmLAOLMLQJ/lNGu0E2ozKDJoMIgqIOy0wiIiZ/MIOfLxzD7B74cHfvu+eezx3DUg4HOEMx4WM4xgXGBw1QxxIm0Ej2N1CJy9ZT40iMe3jFE5axFXwfN/jCAnYGW73GJlbQS/g91rCKh7/e+IJqtPSJOUyjjxlM4hvlqMt8GOfxHvMr1GLdxXPw2+B9TI0G/EjSFyOljDe0gtf8ft5sdv4I7cTYwGMkNBO+jbv4j104jMqUwzlOIyFTASfooOTf9QMT5SpplgZyIQAAAABJRU5ErkJggg==\')';
+
+function initializeToolbar(callback1 = () => {}, callback2 = () => {}) {
+  $([IPython.events]).on('kernel_ready.Kernel', function() {
+    injectNotebookId();
+  });
+
+  /* We cannot embed button via react, so run custom initialization. */
+  const buttonGroup = $(Jupyter.toolbar.add_buttons_group([
+    Jupyter.keyboard_manager.actions.register ({
+      help   : 'Connect to Neptune',
+      icon   : 'fa-check',
+      handler: callback1,
+    }, 'neptune_notebooks', 'neptune_notebook'),
+    Jupyter.keyboard_manager.actions.register ({
+      help   : 'Upload to Neptune',
+      icon   : 'fa-cloud-upload',
+      label  : 'Upload',
+      handler: callback2,
+    }, 'neptune_notebooks', 'neptune_notebook_upload')
+  ]));
+
+  const neptuneButton = $(buttonGroup.find('button')[0]);
+  const uploadButton = $(buttonGroup.find('button')[1]);
+
+  /* Initialize main neptune button with neptune logo. */
+  neptuneButton
+    .attr('id', 'neptune-authorize-btn')
+    .css('background', NEPTUNE_LOGO_URL + ' 7px 4px no-repeat')
+    .css('min-width', '30px')
+    .find('i')
+    .removeClass('fa-check');
+
+  /* Initialize upload button. */
+  uploadButton
+    .attr('id', 'neptune-upload-btn')
+    .append(' Upload');
+
+  if (!window.localStorage.getItem('neptune_api_token') || !getNotebookId()) {
+    neptuneButton
```

Comment:
we should declare states of those buttons, and change them according to props in componentDidMount and componentDidUpdate.